pkg/cloud/openstack/services/compute/instance.go

```diff
@@ -530,3 +538,45 @@ func (is *Service) UpdateToken() error {
 	}
 	return nil
 }
+
+func (Service) buildSchedulerHints(config *openstackconfigv1.OpenstackProviderSpec, opts servers.CreateOptsBuilder) (servers.CreateOptsBuilder, error) {
+	if config.SchedulerHints == nil {
+		return opts, nil
+	}
+	hints := queryFixer{
+		SchedulerHints: schedulerhints.SchedulerHints{
+			Group:           config.SchedulerHints.Group,
+			DifferentHost:   config.SchedulerHints.DifferentHost,
+			SameHost:        config.SchedulerHints.SameHost,
+			TargetCell:      config.SchedulerHints.TargetCell,
+			BuildNearHostIP: config.SchedulerHints.BuildNearHostIP,
```

Comment:
Why are we not using the Query field? https://github.com/gophercloud/gophercloud/blob/19abc56a8cd8d3630ab9735173cb92c0bf635f33/openstack/compute/v2/extensions/schedulerhints/requests.go#L28

Reply:
seems the query is listed below?

Reply:
Yup but it's build custom on top and (at least the current) struct directly has it. So seems like this workaround is not requrired anymore